src/ec/packages/ec-component-page-banner/page-banner.story.js

```diff
@@ -1,65 +1,161 @@
+let leftData = { ...bannerDataAlignLeft };
+
+// Show/hide buttons for optional elements.
+const defaultDescBtnToggler = () => {
+  defaultData.baseline = defaultData.baseline
+    ? false
+    : bannerDataDefault.baseline;
+};
+const imageDescBtnToggler = () => {
+  imageData.baseline = imageData.baseline ? false : bannerDataImage.baseline;
+};
+const imageShadeDescBtnToggler = () => {
+  imageShadeData.baseline = imageShadeData.baseline
+    ? false
+    : bannerDataImageShade.baseline;
+};
+const primaryDescBtnToggler = () => {
+  primaryData.baseline = primaryData.baseline
+    ? false
+    : bannerDataPrimary.baseline;
+};
+const leftDescBtnToggler = () => {
+  leftData.baseline = leftData.baseline ? false : bannerDataAlignLeft.baseline;
+};
+const defaultTitleBtnToggler = () => {
+  defaultData.title = defaultData.title ? false : bannerDataDefault.title;
+};
+const imageTitleBtnToggler = () => {
+  imageData.title = imageData.title ? false : bannerDataImage.title;
+};
+const imageShadeTitleBtnToggler = () => {
+  imageShadeData.title = imageShadeData.title
+    ? false
+    : bannerDataImageShade.title;
+};
+const primaryTitleBtnToggler = () => {
+  primaryData.title = primaryData.title ? false : bannerDataPrimary.title;
+};
+const leftTitleBtnToggler = () => {
+  leftData.title = leftData.title ? false : bannerDataAlignLeft.title;
+};
+const defaultCtaBtnToggler = () => {
+  defaultData.link = defaultData.link ? false : bannerDataDefault.link;
+};
+const imageCtaBtnToggler = () => {
+  imageData.link = imageData.link ? false : bannerDataImage.link;
+};
+const imageShadeCtaBtnToggler = () => {
+  imageShadeData.link = imageShadeData.link ? false : bannerDataImage.link;
+};
+const primaryCtaBtnToggler = () => {
+  primaryData.link = primaryData.link ? false : bannerDataPrimary.link;
+};
+const leftCtaBtnToggler = () => {
+  leftData.link = leftData.link ? false : bannerDataAlignLeft.link;
+};
+// Reset buttons.
+const defaultResetBtnToggler = () => {
+  defaultData = { ...bannerDataDefault };
+};
+const imageResetBtnToggler = () => {
+  imageData = { ...bannerDataImage };
+};
+const imageShadeResetBtnToggler = () => {
+  imageShadeData = { ...bannerDataImageShade };
+};
+const primaryResetBtnToggler = () => {
+  primaryData = { ...bannerDataPrimary };
+};
+const leftResetBtnToggler = () => {
+  leftData = { ...bannerDataAlignLeft };
+};
+
 uiIcons.unshift('null');
 const preparePageBanner = data => {
   if (data.centered) {
     data.centered = boolean('centered', data.centered, tabLabels.states);
   }
 
   data.type = select('type', [data.type], data.type, tabLabels.required);
-  data.title = text('title', data.title, tabLabels.required);
+  if (data.title) {
+    data.title = text('title', data.title, tabLabels.optional);
+  }
   if (data.baseline) {
-    data.baseline = text('baseline', data.baseline, tabLabels.required);
+    data.baseline = text('baseline', data.baseline, tabLabels.optional);
```

Comment:
What property is optional and what is required is a mystery. How do you decide this?
Group of mandatory for primary, default and alight-left knobs don't make sense to me. We had had a chat about this in slack, but here are my thoughts again:
- having a dropdown select in 1 element doesn't make sense. If there is 1 value, it's not an option, no need to be in the storybook UI.
- having a dropdown with a string being an image path doesn't make sense. If it's a text field which can be modified, then it's a real knob and it will have a value of existence in the storybook UI.
- displaying type knob in this case is unnecessary. If there was 1 view where the UI can really change based on this type, ok, but this is not the case.

Reply:
> What property is optional and what is required is a mystery. How do you decide this

Well, the primary source is the ECL usage page of a component, i'm not deciding myself what is optional or mandatory, ther are grey areas, though, and i think this is also a good way to discover them, sometimes other considerations have been used to compensate these grey areas, for this i use the W3c for references around the validity of html elements.

`We had had a chat about this in slack, but here are my thoughts again:`

But i haven't forgotten your remarks, @kalinchernev, i tried to explain the reason why of these choices and i cannot necessarily convince you about them but i cannot even assume that a set of consistent choices has to be changed because of your remarks.

Most of the components use icons, for instance, and it's crucial in this case to pass the icon path to the component to make it work, so if we wanted to follow all the data through the knobs, the icon_path should not be skipped and having it as a text field would just give the users the chance to breake the demo.

Also the `type` is an example of something we already encountered, it controls the variant of the component, it's an important piece of data to be shown but we have different demos for all the variants, so if we give the chance to select the other ones the user would basically reproduce an already existing example in the demo, just that.

Having said this, it would come easy to turn the icon_path, or logo_path into textfields and the variants to be selectable, if that's what you want, but i don't think it would be better, i wouldn't have looked for this solution ottherwise.

Reply:
I've created https://citnet.tech.ec.europa.eu/CITnet/jira/browse/FRONT-1159 
I am starting to get the feeling knobs are so sophisticated part of the ecl-twig project that the components are a minor detail around the core logic of displaying many knobs on each property regardless needs.